src/pydicom/filewriter.py

```diff
@@ -1259,7 +1264,7 @@ def dcmwrite(
     cls_name = dataset.__class__.__name__
 
     # Check for disallowed tags
-    bad_tags = [x >> 16 for x in dataset._dict if x >> 16 in (0, 2)]
+    bad_tags = [group for x in dataset._dict if (group := x >> 16) in (0, 2)]
```

Comment:
Is this actually faster?  Seems like assigning the value to `group` for every tag would add up more than the rare times the op is done again (and if there then on our way to a `raise` so time doesn't matter).

Also (time not mattering as above, but conceptually simpler), could just use a set rather than a list because only the existence of the group is used or not, not individual tags, ... and so it could really be a set called `bad_groups`.

Reply:
I didn't actually check, but you're probably right. Probably only more efficient when the evaluation is expensive. I'll change it back